mongo/collection.go

```diff
@@ -1275,8 +1398,20 @@ func (coll *Collection) FindOneAndReplace(ctx context.Context, filter interface{
 	return coll.findAndModify(ctx, op)
 }
 
-// FindOneAndUpdate finds a single document and updates it, returning either
-// the original or the updated.
+// FindOneAndUpdate executes a findAndModify operation
+// (https://docs.mongodb.com/manual/reference/command/findAndModify/) to update at most one document in the collection.
+//
+// The filter parameter should be a document containing query operators and can be used to select the document to be
+// updated. It cannot be nil. An empty document (e.g. bson.D{}) will select one of the documents in the collection. If
+// the filter does not match any documents, a SingleResult with an error set to ErrNoDocuments wil be returned. If the
+// filter matches multiple documents, one will be selected from the matched set.
```

Comment:
Ditto select and sort.

Reply:
resolving so we can continue conversation in the `Find` thread.